internal/forms/forms.go

```diff
@@ -709,30 +710,28 @@ func (m *Manager) findAbsPathForTemplatePath(tmplPath string) (string, error) {
 	return retVal, nil
 }
 
+// gpsPos returns the current GPS Position
 func (m *Manager) gpsPos() (gpsd.Position, error) {
-	// return current GPS Position
-
 	addr := m.config.GPSd.Addr
-	if addr != "" {
-		conn, err := gpsd.Dial(addr)
-		if err != nil {
-			log.Printf("GPSd daemon: %s", err)
-		}
-		defer conn.Close()
+	if addr == "" {
+		return gpsd.Position{}, errors.New("GPSd: not configured.")
+	}
 
-		conn.Watch(true)
+	conn, err := gpsd.Dial(addr)
 
-		log.Println("Waiting for position from GPSd...")
-		pos, err := conn.NextPosTimeout(3 * time.Second) // TODO: configurable
-		if err != nil {
-			log.Printf("GPSd: %s", err)
-		} else {
-			log.Printf("GPSd: %f %f", pos.Lat, pos.Lon)
-		}
-		return pos, err
-	} else {
-		return gpsd.Position{}, errors.New("GPSd: not configured.")
+	defer conn.Close()
```

Comment:
Very dangerous to defer conn.Close before checking that error. If there is an error, changes are `conn` is nil and you will get a nil pointer panic once the deferred Close call is executed 😅